src/matcher.ts

```diff
@@ -85,6 +85,9 @@ function _matchRoutes(path: string, table: RouteTable): RadixNodeData[] {
   // Order should be from less specific to most specific
   const matches = [];
 
+  // Ignore query parameters
+  path = path.split("?")[0];
```

Comment:
We need to use index check (like https://github.com/unjs/h3/blob/main/src/router.ts#L86) to handle URL like `/login?returnTo=/path?foo=bar` (context: https://github.com/unjs/h3/issues/190)